zulipterminal/ui_tools/views.py

```diff
@@ -945,40 +944,26 @@ def __init__(self, controller: Any, color: str, desc: str,
 
 
 class MsgInfoView(PopUpView):
-    def __init__(self, controller: Any, msg: Message) -> None:
+    def __init__(self, controller: Any, msg: Message,
+                 popup_title: str) -> None:
         self.msg = msg
 
+        msg_info = [
+            ('', [('Date & Time', time.ctime(msg['timestamp'])[:-5]),
+                  ('Sender', msg['sender_full_name']),
+                  ('Sender\'s Email ID', msg['sender_email'])]),
+        ]
         if msg['reactions']:
             reactions = sorted(
-                        [reaction['emoji_name']
-                         + ": "
-                         + reaction['user']['full_name']
-                         + "\n"
-                         for reaction in msg['reactions']])
-            reactions[-1] = reactions[-1].rstrip("\n")
-
-        msg_info = OrderedDict([
-            ('Date & Time', time.ctime(msg['timestamp'])[:-5]),
-            ('Sender', msg['sender_full_name']),
-            ('Sender\'s Email ID', msg['sender_email']),
-            ('Reactions', reactions if msg['reactions'] else '---None---'),
-            ])
-
-        widths = [(len(field) + 7,
-                  max(len(reaction_users) for reaction_users in data)
-                  if isinstance(data, list)
-                  else len(data) + 2)
-                  for field, data in msg_info.items()]
-        max_widths = [max(width) for width in zip(*widths)]
-        self.width = sum(max_widths)
-        self.height = len(msg_info['Reactions']) + 4 if msg['reactions'] else 5
-
-        msg_info_content = [urwid.AttrWrap(
-                urwid.Columns([
-                    urwid.Text(field),
-                    (max_widths[1], urwid.Text(data))
-                ], dividechars=2),
-                None if index % 2 else 'popup_contrast')
-             for index, (field, data) in enumerate(msg_info.items())]
+                (reaction['emoji_name'], reaction['user']['full_name'])
+                for reaction in msg['reactions']
+            )
+            msg_info.append(('Reactions', reactions))
```

Comment:
I like the reactions in a different category - that seems clearer for sure - but what do you think of grouping responses by alternating color line? Currently (in 0.5.1) it's all one color, which is the other extreme. I see we could do this by reaction, or *maybe* even by user? Grouping would also mean we could avoid having 'tada' listed however many times, for example!

If this fits within the current feature of the revised popup code, we could make that change in this commit, but otherwise we could delay this until later.

Reply:
@neiljp With the current (PR) code, one way to group could be using the reaction name for the first reactor and then leaving the subsequent left column blank for other reactors (with alternating `popup_contrast` style).

For instance,
```
+1       User 1 (popup_contrast)
         User 2
         User 3 (popup_contrast)
tada     User 1 
         User 2 (popup_contrast)
         User 3 

```
```python
[('+1', 'User 1'), ('', 'User 2'), ('', 'User 3'), ('tada', 'User 1'), ('', 'User 2'), ('', 'User 3')]
```


Does the proposed styling look good or should I make amendments to the `make_table_with_categories` code to incorporate the other behaviour (grouping responses by alternating colour line) that you suggested?

Reply:
This is one option. I think originally I was considering you could have the users joined with newlines, so the above would be more like:
```
+1    User 1 (popup_contrast)
      User 2 (popup_contrast)
      User 3 (popup_contrast)
tada  User 1
      User 2
      User 3
smile User 4 (popup_contrast)
```
I thought this might use the contrast to group the reactions better, perhaps.